Libplanet/Store/StoreExtension.cs

```diff
@@ -18,14 +18,15 @@ public static class StoreExtension
         /// <param name="address">The <see cref="Address"/> to look up.</param>
         /// <param name="lookupUntil">The upper bound (i.e., the latest block) of the search range.
         /// <see cref="Block{T}"/>s after <paramref name="lookupUntil"/> are ignored.</param>
-        /// <returns>A <see cref="Block{T}.Hash"/> which has the state of
-        /// the <paramref name="address"/>.</returns>
+        /// <returns>Returns a nullable tuple consisting of <see cref="Block{T}.Hash"/> and
+        /// <see cref="Block{T}.Index"/> of the <see cref="Block{T}"/> with the state of the
+        /// address.</returns>
         /// <typeparam name="T">An <see cref="IAction"/> class used with
         /// <paramref name="lookupUntil"/>.</typeparam>
         /// <seealso
         /// cref="IStore.StoreStateReference{T}(string, IImmutableSet{Address}, Block{T})"/>
         /// <seealso cref="IStore.IterateStateReferences(string, Address)"/>
-        public static HashDigest<SHA256>? LookupStateReference<T>(
+        public static Tuple<HashDigest<SHA256>, long> LookupStateReference<T>(
```

Comment:
How about making this method `internal` and renaming to like `LookupStateReferenceWithIndex<T>()`, and then adding a wrapper method named `LookupStateReference<T>()` around it in order to maintain backward compatibility?

Reply:
Elaboration: This makes a backward-incompatible change on the public interface, which should increase a minor version (not a patch version), according to the Semantic Versioning.  However, this kind of problem is definitely not about usability nor feature lacking, but a bug, so it should be a patch release as possible.  So we'd better to add a new internal method (`LookupStateReferenceWithIndex<T>()`) in this pull request, and then replace the old method (`LookupStateReference<T>()`) with the new one (`LookupStateReferenceWithIndex<T>()`) in a separated pull request which targets to the master branch.

Reply:
I added `LookupStateReferenceWithIndex<T>()` method and maintained `LookupStateReference<T>()` interface as you suggested.